src/DeviceInterface/initFormSubmissionsApi.js

```diff
@@ -50,11 +51,15 @@ export function initFormSubmissionsApi (forms, matching) {
             const button = /** @type HTMLElement */(event.target)?.closest(selector)
             if (!button) return
 
+            // If the element we've found includes a form it can't be a button, it's a false match
+            const buttonIsAFalsePositive = formsArray.some(form => button?.contains(form.form))
+            if (buttonIsAFalsePositive) return
```

Comment:
This fixes an issue where we could prompt to save prematurely if a form is descendant of an element with a class of `.button` (wtf, right?) and the user perform a specific sequence of actions. It's rather complex to test, so I left that out. Let me know if you disagree.